components/vulkan/include/components/vulkan/common/structure_type_helpers.hpp

```diff
@@ -0,0 +1,3882 @@
+{
+	return VK_STRUCTURE_TYPE_PIPELINE_VIEWPORT_STATE_CREATE_INFO;
+}
+template <>
+VkStructureType get_structure_type<VkPipelineRasterizationStateCreateInfo>()
+{
+	return VK_STRUCTURE_TYPE_PIPELINE_RASTERIZATION_STATE_CREATE_INFO;
+}
+template <>
+VkStructureType get_structure_type<VkPipelineMultisampleStateCreateInfo>()
+{
+	return VK_STRUCTURE_TYPE_PIPELINE_MULTISAMPLE_STATE_CREATE_INFO;
+}
+template <>
+VkStructureType get_structure_type<VkPipelineDepthStencilStateCreateInfo>()
+{
+	return VK_STRUCTURE_TYPE_PIPELINE_DEPTH_STENCIL_STATE_CREATE_INFO;
+}
+template <>
+VkStructureType get_structure_type<VkPipelineColorBlendStateCreateInfo>()
+{
+	return VK_STRUCTURE_TYPE_PIPELINE_COLOR_BLEND_STATE_CREATE_INFO;
+}
+template <>
+VkStructureType get_structure_type<VkPipelineDynamicStateCreateInfo>()
+{
+	return VK_STRUCTURE_TYPE_PIPELINE_DYNAMIC_STATE_CREATE_INFO;
+}
+template <>
+VkStructureType get_structure_type<VkGraphicsPipelineCreateInfo>()
+{
+	return VK_STRUCTURE_TYPE_GRAPHICS_PIPELINE_CREATE_INFO;
+}
+template <>
+VkStructureType get_structure_type<VkPipelineLayoutCreateInfo>()
+{
+	return VK_STRUCTURE_TYPE_PIPELINE_LAYOUT_CREATE_INFO;
+}
+template <>
+VkStructureType get_structure_type<VkSamplerCreateInfo>()
+{
+	return VK_STRUCTURE_TYPE_SAMPLER_CREATE_INFO;
+}
+template <>
+VkStructureType get_structure_type<VkCopyDescriptorSet>()
+{
+	return VK_STRUCTURE_TYPE_COPY_DESCRIPTOR_SET;
+}
+template <>
+VkStructureType get_structure_type<VkDescriptorPoolCreateInfo>()
+{
+	return VK_STRUCTURE_TYPE_DESCRIPTOR_POOL_CREATE_INFO;
+}
+template <>
+VkStructureType get_structure_type<VkDescriptorSetAllocateInfo>()
+{
+	return VK_STRUCTURE_TYPE_DESCRIPTOR_SET_ALLOCATE_INFO;
+}
+template <>
+VkStructureType get_structure_type<VkDescriptorSetLayoutCreateInfo>()
+{
+	return VK_STRUCTURE_TYPE_DESCRIPTOR_SET_LAYOUT_CREATE_INFO;
+}
+template <>
+VkStructureType get_structure_type<VkWriteDescriptorSet>()
+{
+	return VK_STRUCTURE_TYPE_WRITE_DESCRIPTOR_SET;
+}
+template <>
+VkStructureType get_structure_type<VkFramebufferCreateInfo>()
+{
+	return VK_STRUCTURE_TYPE_FRAMEBUFFER_CREATE_INFO;
+}
+template <>
+VkStructureType get_structure_type<VkRenderPassCreateInfo>()
+{
+	return VK_STRUCTURE_TYPE_RENDER_PASS_CREATE_INFO;
+}
+template <>
+VkStructureType get_structure_type<VkCommandPoolCreateInfo>()
+{
+	return VK_STRUCTURE_TYPE_COMMAND_POOL_CREATE_INFO;
+}
+template <>
+VkStructureType get_structure_type<VkCommandBufferAllocateInfo>()
+{
+	return VK_STRUCTURE_TYPE_COMMAND_BUFFER_ALLOCATE_INFO;
+}
+template <>
+VkStructureType get_structure_type<VkCommandBufferInheritanceInfo>()
+{
+	return VK_STRUCTURE_TYPE_COMMAND_BUFFER_INHERITANCE_INFO;
+}
+template <>
+VkStructureType get_structure_type<VkCommandBufferBeginInfo>()
+{
+	return VK_STRUCTURE_TYPE_COMMAND_BUFFER_BEGIN_INFO;
+}
+template <>
+VkStructureType get_structure_type<VkRenderPassBeginInfo>()
+{
+	return VK_STRUCTURE_TYPE_RENDER_PASS_BEGIN_INFO;
+}
+
```

Comment:
Where do the empty line before all the `#endif`s come from?

Reply:
Not sure. It's not obvious looking at the generator.

I can run clang format on the generated file to try and clean it up a bit

Reply:
Fixed, was in the generator helper when processing sections